### DISCUSSION TITLE
how to use tcd instead of lcd/cd to maintain one context per tab

### DISCUSSION THREAD

**Discussion by alexmaloteaux** [nvim-tree/nvim-tree.lua#1847] [Q&A] [answered]
Hi
im most of the time developping multiple project at the same time, for this pattern im using one tab per project and all my telescope..... configs are all related to cwd. As a maybe bad comparison this allows me to have like goland idea in one tab, pycharm in another ... 

atm im still using nerdtree, because this one allow to use tcd and have completly separate tabs tree.
If not mistaken , nvim-tree doesnt allow this specific configuration or if it does , could somebody explain me how to achieve this.

the best for me would be that ctrl+] change the root of nvim-tree, change cwd with tcd and of course the root for lsp. and that all tabs are completly "isolated", at least as much as neovim allows, the rest relying on specific plugin config 

Best Regards

**Comment by alex-courtis** [comment, marked as answer]
nvim-tree doesn't really support different tabs per tree. It simply shows the one tree in different tab windows.

Multiple instances would be the solution here. See #457

**Comment by aerosol** [comment]
nvim-tree doesn't need to display multiple instances -- one tree is fine; Unless I'm completely misinterpreting this, I think the point here is being able to use `tcd` on `C-]`, that's it. As I understand it, it currently allows you to choose between `cd` and `lcd`. Would `tcd` option break anything?

**Reply by alex-courtis** [reply]
> nvim-tree doesn't need to display multiple instances -- one tree is fine; Unless I'm completely misinterpreting this, I think the point here is being able to use `tcd` on `C-]`, that's it. As I understand it, it currently allows you to choose between `cd` and `lcd`. Would `tcd` option break anything?

That is quite insightful and may go a long way towards multi-tab support.

`lcd` (sync_root_with_cwd false) mostly works with some issues around the tree open state being lazily propagated between tabs.

`cd` (sync_root_with_cwd true) is less functional, with tree state being overridden (closed) following tab switch.

**Comment by alex-courtis** [comment]
Tracking at
- #2255
